### ISSUE TITLE
Error when building: Data did not validate against the supplied schema

### ISSUE THREAD

**Issue by OlgaMaslova** [aws-amplify/amplify-cli#8875]
**Please describe which feature you have a question about?**
My build fails with the following error:
`There was an error initializing your environment.
2021-11-15T13:29:17.531Z [INFO]: [0mError: Data did not validate against the supplied schema. Underlying errors were [{"keyword":"required","dataPath":"","schemaPath":"#/required","params":{"missingProperty":"storageAccess"},"message":"should have required property 'storageAccess'"}]`

**Provide additional details**
Amplify push from amlify-cli builds the backend. However when the code is pushed to gitlab there is this error. No idea what happened. No changes were brought to this commit in the backend.

**What AWS Services are you utilizing?**
Amplify console. **AppID**: do5lfh5kuwuas, **region**: eu-west-1.
Amplify cli is set to latest, so 7.3.1. I have just checked, no problem when build with 6.4.0.

**Provide additional details e.g. code snippets**
Image
![image](https://user-images.githubusercontent.com/29795366/141791093-e7a54232-2991-4e25-a25c-e6f2a959b1ef.png)

**Comment by kaustavghosh06** [comment]
Thanks @OlgaMaslova We have enough information to investigate at this point. We'll investigate right away and get back to you.

**Comment by dmost714** [comment]
I may be encountering a similar issue.
All files are commited, and I `amplify pull -y`, and `amplify status` shows no changes.
Then I make any change to my model (e.g. just add another return char at the end of the file).
When I `amplify push -y` it fails with a similar error to the one above:
```➜  iwpk-portal git:(dev) ✗ amplify push -y
✔ Successfully pulled backend environment dev from the cloud.

    Current Environment: dev
    
┌──────────┬────────────────────────────────────────────┬───────────┬───────────────────┐
│ Category │ Resource name                              │ Operation │ Provider plugin   │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Api      │ iwpkportal                                 │ Update    │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Function │ iwpkportal2735839927358399PostConfirmation │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Function │ iwpkportal2735839927358399PreSignup        │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Function │ iwpkGQLResolver                            │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Function │ S3Trigger8d374d54                          │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Auth     │ userPoolGroups                             │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Auth     │ iwpkportal2735839927358399                 │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Storage  │ s3iwpkassets                               │ No Change │ awscloudformation │
└──────────┴────────────────────────────────────────────┴───────────┴───────────────────┘
✖ An error occurred when pushing the resources to the cloud
🛑 An error occurred during the push operation: Data did not validate against the supplied schema. Underlying errors were [{"keyword":"required","dataPath":"","schemaPath":"#/required","params":{"missingProperty":"storageAccess"},"message":"should have required property 'storageAccess'"}]
➜  iwpk-portal git:(dev) ✗ amplify status

    Current Environment: dev
    
┌──────────┬────────────────────────────────────────────┬───────────┬───────────────────┐
│ Category │ Resource name                              │ Operation │ Provider plugin   │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Auth     │ userPoolGroups                             │ Update    │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Auth     │ iwpkportal2735839927358399                 │ Update    │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Api      │ iwpkportal                                 │ Update    │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Function │ iwpkportal2735839927358399PostConfirmation │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Function │ iwpkportal2735839927358399PreSignup        │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Function │ iwpkGQLResolver                            │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Function │ S3Trigger8d374d54                          │ No Change │ awscloudformation │
├──────────┼────────────────────────────────────────────┼───────────┼───────────────────┤
│ Storage  │ s3iwpkassets                               │ No Change │ awscloudformation │
└──────────┴────────────────────────────────────────────┴───────────┴───────────────────┘
```
And after, there are multiple files modified and `amplify status` shows multiple categories with changes.
Some of those changes appear related to an auth group change (deleted one group and added a new/different one).

Here is my schema
```
type PromotionalKit
  @model
  @key(name: "byStatus", fields: ["status", "updatedAt"])
  @auth(
    rules: [
      { allow: groups, groups: ["ADMIN","AGENT"], queries: [get, list], mutations: [create] }
      { allow: owner, queries: [get, list], mutations: [create] }
    ]
  ) {
  id: ID!
  createdAt: AWSDateTime
  updatedAt: AWSDateTime
  user_identity_id: String
  status: String
  quantityOrdered: Int
  quantityProduced: Int
}

type Mutation {
  authorizeUser(userName: String, hash: String): AWSJSON
    @function(name: "iwpkGQLResolver-${env}"
  )
  processKitOrder(orderId: String): AWSJSON
    @function(name: "iwpkGQLResolver-${env}"
  )
}
```

I'm running Amplify CLI v7.3.1.

**Comment by kaustavghosh06** [comment]
@dmost714 Could you please tell us the JSON in  `amplify/backend/storage/parameters.json` and  `amplify/backend/storage/s3-cloudformation.json`?

**Comment by dmost714** [comment]
After performing `amplify pull -y` and having no changed files:
```
➜  iwpk-portal git:(dev) ✗ amplify storage update
? Select from one of the below mentioned services: Content (Images, audio, video, etc.)
✔ A migration is needed to support latest updates on storage resources.
Recommended to try in a non-production environment first. Run "amplify env add" to create or clone an environment.
Learn more about this migration: https://docs.amplify.aws/cli/migration/override
Do you want to migrate storage resource "s3iwpkassets"? (Y/n) · yes
✔ A migration is needed to support latest updates on auth resources.
Recommended to try in a non-production environment first. Run "amplify env add" to create or clone an environment.
Learn more about this migration: https://docs.amplify.aws/cli/migration/override
Do you want to migrate auth resource "iwpkportal2735839927358399"? (Y/n) · yes
Error: Data did not validate against the supplied schema. Underlying errors were [{"keyword":"required","dataPath":"","schemaPath":"#/required","params":{"missingProperty":"storageAccess"},"message":"should have required property 'storageAccess'"}]
    at CLIInputSchemaValidator.validateInput (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/src/category-interfaces/category-base-schema-generator.ts:133:13)
    at S3InputState.isCLIInputsValid (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts:381:12)
    at S3InputState.saveCliInputPayload (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts:538:5)
    at S3InputState.migrate (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts:265:5)
    at updateWalkthrough (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.ts:147:9)
    at Object.executeAmplifyCommand (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-storage/src/index.ts:185:3)
    at executePluginModuleCommand (/usr/local/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:204:3)
    at Object.executeCommand (/usr/local/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:30:5)
    at Object.run (/usr/local/lib/node_modules/@aws-amplify/cli/src/index.ts:205:5)
🛑 An error occurred when updating the storage resource
➜  iwpk-portal git:(dev) ✗ 
```

I'll try creating a new env as suggested.

**Comment by dmost714** [comment]
**/amplify/backend/storage/s3iwpkassets/parameters.json**
```
{
  "bucketName": "iwpkportal-assets",
  "authPolicyName": "s3_amplify_45b60f6e",
  "unauthPolicyName": "s3_amplify_45b60f6e",
  "authRoleName": {
    "Ref": "AuthRoleName"
  },
  "unauthRoleName": {
    "Ref": "UnauthRoleName"
  },
  "selectedGuestPermissions": [
    "s3:GetObject",
    "s3:ListBucket"
  ],
  "selectedAuthenticatedPermissions": [
    "s3:GetObject",
    "s3:ListBucket"
  ],
  "triggerFunction": "NONE",
  "s3PermissionsGuestPublic": "DISALLOW",
  "s3PermissionsGuestUploads": "DISALLOW",
  "GuestAllowList": "DISALLOW",
  "s3PermissionsAuthenticatedPublic": "DISALLOW",
  "s3PermissionsAuthenticatedProtected": "DISALLOW",
  "s3PermissionsAuthenticatedPrivate": "DISALLOW",
  "s3PermissionsAuthenticatedUploads": "DISALLOW",
  "AuthenticatedAllowList": "DISALLOW"
}
```

**/amplify/backend/storage/s3iwpkassets/s3-cloudformation-template.json**
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "S3 resource stack creation using Amplify CLI",
  "Parameters": {
    "bucketName": {
      "Type": "String"
    },
    "authPolicyName": {
      "Type": "String"
    },
    "unauthPolicyName": {
      "Type": "String"
    },
    "authRoleName": {
      "Type": "String"
    },
    "unauthRoleName": {
      "Type": "String"
    },
    "s3PublicPolicy": {
      "Type": "String",
      "Default": "NONE"
    },
    "s3PrivatePolicy": {
      "Type": "String",
      "Default": "NONE"
    },
    "s3ProtectedPolicy": {
      "Type": "String",
      "Default": "NONE"
    },
    "s3UploadsPolicy": {
      "Type": "String",
      "Default": "NONE"
    },
    "s3ReadPolicy": {
      "Type": "String",
      "Default": "NONE"
    },
    "s3PermissionsAuthenticatedPublic": {
      "Type": "String",
      "Default": "DISALLOW"
    },
    "s3PermissionsAuthenticatedProtected": {
      "Type": "String",
      "Default": "DISALLOW"
    },
    "s3PermissionsAuthenticatedPrivate": {
      "Type": "String",
      "Default": "DISALLOW"
    },
    "s3PermissionsAuthenticatedUploads": {
      "Type": "String",
      "Default": "DISALLOW"
    },
    "s3PermissionsGuestPublic": {
      "Type": "String",
      "Default": "DISALLOW"
    },
    "s3PermissionsGuestUploads": {
      "Type": "String",
      "Default": "DISALLOW"
    },
    "AuthenticatedAllowList": {
      "Type": "String",
      "Default": "DISALLOW"
    },
    "GuestAllowList": {
      "Type": "String",
      "Default": "DISALLOW"
    },
    "selectedGuestPermissions": {
      "Type": "CommaDelimitedList",
      "Default": "NONE"
    },
    "selectedAuthenticatedPermissions": {
      "Type": "CommaDelimitedList",
      "Default": "NONE"
    },
    "env": {
      "Type": "String"
    },
    "triggerFunction": {
      "Type": "String"
    },
    "authiwpkportal2735839927358399UserPoolId": {
      "Type": "String",
      "Default": "authiwpkportal2735839927358399UserPoolId"
    },
    "authuserPoolGroupsINTOXALOCKGroupRole": {
      "Type": "String",
      "Default": "authuserPoolGroupsINTOXALOCKGroupRole"
    },
    "authuserPoolGroupsFULFILLMENTGroupRole": {
      "Type": "String",
      "Default": "authuserPoolGroupsFULFILLMENTGroupRole"
    }
  },
  "Conditions": {
    "ShouldNotCreateEnvResources": {
      "Fn::Equals": [
        {
          "Ref": "env"
        },
        "NONE"
      ]
    },
    "CreateAuthPublic": {
      "Fn::Not": [
        {
          "Fn::Equals": [
            {
              "Ref": "s3PermissionsAuthenticatedPublic"
            },
            "DISALLOW"
          ]
        }
      ]
    },
    "CreateAuthProtected": {
      "Fn::Not": [
        {
          "Fn::Equals": [
            {
              "Ref": "s3PermissionsAuthenticatedProtected"
            },
            "DISALLOW"
          ]
        }
      ]
    },
    "CreateAuthPrivate": {
      "Fn::Not": [
        {
          "Fn::Equals": [
            {
              "Ref": "s3PermissionsAuthenticatedPrivate"
            },
            "DISALLOW"
          ]
        }
      ]
    },
    "CreateAuthUploads": {
      "Fn::Not": [
        {
          "Fn::Equals": [
            {
              "Ref": "s3PermissionsAuthenticatedUploads"
            },
            "DISALLOW"
          ]
        }
      ]
    },
    "CreateGuestPublic": {
      "Fn::Not": [
        {
          "Fn::Equals": [
            {
              "Ref": "s3PermissionsGuestPublic"
            },
            "DISALLOW"
          ]
        }
      ]
    },
    "CreateGuestUploads": {
      "Fn::Not": [
        {
          "Fn::Equals": [
            {
              "Ref": "s3PermissionsGuestUploads"
            },
            "DISALLOW"
          ]
        }
      ]
    },
    "AuthReadAndList": {
      "Fn::Not": [
        {
          "Fn::Equals": [
            {
              "Ref": "AuthenticatedAllowList"
            },
            "DISALLOW"
          ]
        }
      ]
    },
    "GuestReadAndList": {
      "Fn::Not": [
        {
          "Fn::Equals": [
            {
              "Ref": "GuestAllowList"
            },
            "DISALLOW"
          ]
        }
      ]
    }
  },
  "Resources": {
    "S3Bucket": {
      "Type": "AWS::S3::Bucket",
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketName": {
          "Fn::If": [
            "ShouldNotCreateEnvResources",
            {
              "Ref": "bucketName"
            },
            {
              "Fn::Join": [
                "",
                [
                  {
                    "Ref": "bucketName"
                  },
                  {
                    "Fn::Select": [
                      3,
                      {
                        "Fn::Split": [
                          "-",
                          {
                            "Ref": "AWS::StackName"
                          }
                        ]
                      }
                    ]
                  },
                  "-",
                  {
                    "Ref": "env"
                  }
                ]
              ]
            }
          ]
        },
        "CorsConfiguration": {
          "CorsRules": [
            {
              "AllowedHeaders": [
                "*"
              ],
              "AllowedMethods": [
                "GET",
                "HEAD",
                "PUT",
                "POST",
                "DELETE"
              ],
              "AllowedOrigins": [
                "*"
              ],
              "ExposedHeaders": [
                "x-amz-server-side-encryption",
                "x-amz-request-id",
                "x-amz-id-2",
                "ETag"
              ],
              "Id": "S3CORSRuleId1",
              "MaxAge": "3000"
            }
          ]
        }
      }
    },
    "INTOXALOCKGroupPolicy": {
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": "INTOXALOCK-group-s3-policy",
        "Roles": [
          {
            "Fn::Join": [
              "",
              [
                {
                  "Ref": "authiwpkportal2735839927358399UserPoolId"
                },
                "-INTOXALOCKGroupRole"
              ]
            ]
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": [
                "s3:PutObject"
              ],
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/*"
                    ]
                  ]
                }
              ]
            }
          ]
        }
      }
    },
    "S3AuthPublicPolicy": {
      "DependsOn": [
        "S3Bucket"
      ],
      "Condition": "CreateAuthPublic",
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": {
          "Ref": "s3PublicPolicy"
        },
        "Roles": [
          {
            "Ref": "authRoleName"
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": {
                "Fn::Split": [
                  ",",
                  {
                    "Ref": "s3PermissionsAuthenticatedPublic"
                  }
                ]
              },
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/public/*"
                    ]
                  ]
                }
              ]
            }
          ]
        }
      }
    },
    "S3AuthProtectedPolicy": {
      "DependsOn": [
        "S3Bucket"
      ],
      "Condition": "CreateAuthProtected",
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": {
          "Ref": "s3ProtectedPolicy"
        },
        "Roles": [
          {
            "Ref": "authRoleName"
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": {
                "Fn::Split": [
                  ",",
                  {
                    "Ref": "s3PermissionsAuthenticatedProtected"
                  }
                ]
              },
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/protected/${cognito-identity.amazonaws.com:sub}/*"
                    ]
                  ]
                }
              ]
            }
          ]
        }
      }
    },
    "S3AuthPrivatePolicy": {
      "DependsOn": [
        "S3Bucket"
      ],
      "Condition": "CreateAuthPrivate",
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": {
          "Ref": "s3PrivatePolicy"
        },
        "Roles": [
          {
            "Ref": "authRoleName"
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": {
                "Fn::Split": [
                  ",",
                  {
                    "Ref": "s3PermissionsAuthenticatedPrivate"
                  }
                ]
              },
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/private/${cognito-identity.amazonaws.com:sub}/*"
                    ]
                  ]
                }
              ]
            }
          ]
        }
      }
    },
    "S3AuthUploadPolicy": {
      "DependsOn": [
        "S3Bucket"
      ],
      "Condition": "CreateAuthUploads",
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": {
          "Ref": "s3UploadsPolicy"
        },
        "Roles": [
          {
            "Ref": "authRoleName"
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": {
                "Fn::Split": [
                  ",",
                  {
                    "Ref": "s3PermissionsAuthenticatedUploads"
                  }
                ]
              },
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/uploads/*"
                    ]
                  ]
                }
              ]
            }
          ]
        }
      }
    },
    "S3AuthReadPolicy": {
      "DependsOn": [
        "S3Bucket"
      ],
      "Condition": "AuthReadAndList",
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": {
          "Ref": "s3ReadPolicy"
        },
        "Roles": [
          {
            "Ref": "authRoleName"
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": [
                "s3:GetObject"
              ],
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/protected/*"
                    ]
                  ]
                }
              ]
            },
            {
              "Effect": "Allow",
              "Action": [
                "s3:ListBucket"
              ],
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      }
                    ]
                  ]
                }
              ],
              "Condition": {
                "StringLike": {
                  "s3:prefix": [
                    "public/",
                    "public/*",
                    "protected/",
                    "protected/*",
                    "private/${cognito-identity.amazonaws.com:sub}/",
                    "private/${cognito-identity.amazonaws.com:sub}/*"
                  ]
                }
              }
            }
          ]
        }
      }
    },
    "S3GuestPublicPolicy": {
      "DependsOn": [
        "S3Bucket"
      ],
      "Condition": "CreateGuestPublic",
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": {
          "Ref": "s3PublicPolicy"
        },
        "Roles": [
          {
            "Ref": "unauthRoleName"
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": {
                "Fn::Split": [
                  ",",
                  {
                    "Ref": "s3PermissionsGuestPublic"
                  }
                ]
              },
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/public/*"
                    ]
                  ]
                }
              ]
            }
          ]
        }
      }
    },
    "S3GuestUploadPolicy": {
      "DependsOn": [
        "S3Bucket"
      ],
      "Condition": "CreateGuestUploads",
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": {
          "Ref": "s3UploadsPolicy"
        },
        "Roles": [
          {
            "Ref": "unauthRoleName"
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": {
                "Fn::Split": [
                  ",",
                  {
                    "Ref": "s3PermissionsGuestUploads"
                  }
                ]
              },
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/uploads/*"
                    ]
                  ]
                }
              ]
            }
          ]
        }
      }
    },
    "S3GuestReadPolicy": {
      "DependsOn": [
        "S3Bucket"
      ],
      "Condition": "GuestReadAndList",
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": {
          "Ref": "s3ReadPolicy"
        },
        "Roles": [
          {
            "Ref": "unauthRoleName"
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": [
                "s3:GetObject"
              ],
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/protected/*"
                    ]
                  ]
                }
              ]
            },
            {
              "Effect": "Allow",
              "Action": [
                "s3:ListBucket"
              ],
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      }
                    ]
                  ]
                }
              ],
              "Condition": {
                "StringLike": {
                  "s3:prefix": [
                    "public/",
                    "public/*",
                    "protected/",
                    "protected/*"
                  ]
                }
              }
            }
          ]
        }
      }
    },
    "FULFILLMENTGroupPolicy": {
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyName": "FULFILLMENT-group-s3-policy",
        "Roles": [
          {
            "Fn::Join": [
              "",
              [
                {
                  "Ref": "authiwpkportal2735839927358399UserPoolId"
                },
                "-FULFILLMENTGroupRole"
              ]
            ]
          }
        ],
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": [
                "s3:GetObject",
                "s3:ListBucket"
              ],
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      },
                      "/*"
                    ]
                  ]
                }
              ]
            },
            {
              "Effect": "Allow",
              "Action": "s3:ListBucket",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "S3Bucket"
                      }
                    ]
                  ]
                }
              ]
            }
          ]
        }
      }
    }
  },
  "Outputs": {
    "BucketName": {
      "Value": {
        "Ref": "S3Bucket"
      },
      "Description": "Bucket name for the S3 bucket"
    },
    "Region": {
      "Value": {
        "Ref": "AWS::Region"
      }
    }
  }
}

**Comment by dmost714** [comment]
I created a shiny new env, amplify status shows 'Create' for all categories. I did an amplify push, and it failed with the above error. I tried 'amplify storage update' and it also fails with the exception in my comment above. I hope this helps. Happy to provide any additional details needed.

**Comment by kaustavghosh06** [comment]
@dmost714 yes, we were able to reproduce this issue. Thanks for all the information. We're looking at it right away.

**Comment by jk171505** [comment]
I'm having the same issue. Five days ago last deployment with amplify-cli (6.3.1.) everything worked fine. Today some front-end files were updated, nothing related to amplify and the deployment with amplify-cli 7.3.1 fails with the same error.
Although I noticed one issue few lines above related to `amplify-meta.json`. Here is my build log:
`
2021-11-15T15:55:48.180Z [INFO]: [0mAmplify AppID found: d35xzpo2wk0d0z. Amplify App name is: sunstreetadmin[0m
2021-11-15T15:55:48.231Z [INFO]: [0mBackend environment dev found in Amplify Console app: sunstreetadmin[0m
2021-11-15T15:55:49.758Z [INFO]: [31mError: File at path: '/codebuild/output/src667835481/src/sunstreet-admin/webapp/amplify/backend/amplify-meta.json' does not exist[39m
2021-11-15T15:55:49.884Z [INFO]: [0mSuccessfully generated models. Generated models can be found in /codebuild/output/src667835481/src/sunstreet-admin/webapp/src[0m
2021-11-15T15:55:49.893Z [WARNING]: - Fetching updates to backend environment: dev from the cloud.
2021-11-15T15:55:51.220Z [WARNING]: - Building resource auth/userPoolGroups
2021-11-15T15:55:51.221Z [WARNING]: - Building resource auth/sunstreetadmin2be673282be67328
2021-11-15T15:55:51.223Z [WARNING]: - Building resource storage/s3c856bfc7
2021-11-15T15:55:51.282Z [WARNING]: ✔ Successfully pulled backend environment dev from the cloud.
2021-11-15T15:55:51.514Z [INFO]: 
2021-11-15T15:55:56.651Z [INFO]: [33mNote: It is recommended to run this command from the root of your app directory[39m
2021-11-15T15:56:02.594Z [WARNING]: - Initializing your environment: dev
2021-11-15T15:56:03.865Z [WARNING]: - Building resource auth/userPoolGroups
2021-11-15T15:56:03.866Z [WARNING]: - Building resource auth/sunstreetadmin2be673282be67328
2021-11-15T15:56:03.868Z [WARNING]: - Building resource storage/s3c856bfc7
2021-11-15T15:56:03.868Z [WARNING]: ✔ Initialized provider successfully.
2021-11-15T15:56:48.191Z [WARNING]: - Uploading files...
2021-11-15T15:56:48.260Z [WARNING]: - Uploading files...
2021-11-15T15:57:28.137Z [WARNING]: - Uploading files...
2021-11-15T15:57:28.539Z [WARNING]: - Uploading files...
2021-11-15T15:57:29.489Z [WARNING]: - Uploading files...
2021-11-15T15:57:30.984Z [WARNING]: - Uploading files...
2021-11-15T15:57:32.155Z [WARNING]: - Uploading files...
2021-11-15T15:57:32.670Z [WARNING]: - Uploading files...
2021-11-15T15:57:34.046Z [WARNING]: - Uploading files...
2021-11-15T15:57:34.052Z [WARNING]: - Uploading files...
2021-11-15T15:57:35.317Z [WARNING]: - Uploading files...
2021-11-15T15:57:36.957Z [WARNING]: - Uploading files...
2021-11-15T15:57:37.369Z [WARNING]: - Uploading files...
2021-11-15T15:57:39.605Z [WARNING]: - Uploading files...
2021-11-15T15:57:41.173Z [WARNING]: ✖ An error occurred when pushing the resources to the cloud
2021-11-15T15:57:41.176Z [WARNING]: ✖ There was an error initializing your environment.
2021-11-15T15:57:41.181Z [INFO]: [0mError: Data did not validate against the supplied schema. Underlying errors were [{"keyword":"required","dataPath":"","schemaPath":"#/required","params":{"missingProperty":"storageAccess"},"message":"should have required property 'storageAccess'"}][0m
`

**Comment by kaustavghosh06** [comment]
@jk171505 We've root-caused the issue and working on a fix for this. In the meanwhile if you could provide us with your `amplify/backend/storage/parameters.json` and `amplify/backend/storage/s3-cloudformation-template.json` files, we can validate against our fix and make sure it works for your project.

**Comment by jk171505** [comment]
@kaustavghosh06 sent to amplify-cli@amazon.com.

**Comment by kaustavghosh06** [comment]
Thanks @jk171505.
@dmost714 @jk171505 @OlgaMaslova  We have validated all of our files against our fix here (#8881). We plan to do a release with the fixes later today.

**Comment by dmost714** [comment]
Thank you for the quick turnaround on that issue.
Using v7.3.3 I am able to touch the GQL model and push up changes.

I am encountering cloud formation issues when attempting `amplify auth update`. I'll open a new issue for those.

**Comment by sachscode** [comment]
@dmost714  
1. Could you please share the error stack for amplify auth update. 
2. Please share amplify --version

**Comment by dmost714** [comment]
> @dmost714
> 
> 1. Could you please share the error stack for amplify auth update.
> 2. Please share amplify --version

I was using v7.3.3 when I reported the auth update issue.
There was no cli error stack, it was cloud formation failures.
Here's the ticket I opened: [auth issue #8895](https://github.com/aws-amplify/amplify-cli/issues/8895)
Expand the "details" in the **Log Output** section .

Unfortunately, I rebuilt my backend and can no longer reproduce the issue.
